CMSIS/DSP/Include/arm_math.h

```diff
@@ -357,7 +357,7 @@ extern "C"
 
 
 /* Included for instrinsics definitions */
-#if defined (_MSC_VER ) 
+#if defined (_MSC_VER )
```

Comment:
Well, doesn't directly correspond to this PR but should we discuss to move such compiler dependencies to a specific compiler_msc.h rather than cluttering the math header?

Reply:
This one is very special and just to be able to build Python or Matlab wrapper. It is focusing on being able to compile for x86.

There is no reason to have support for this compiler in a separate file and it would not make sense to have such an header in Core or Core_A.

The problem is that there is no pure C version of all DSP functions. m0 versions are using 3 intrinsics which required a C version to be able to compile the Python or Matlab wrappers.

Reply:
Being able to compile code on host machines (Windows, Linux) is not very special to DSP. One might want to do that for unit testing or perhaps for "simulating" target applications. Not sure what's actually worth to have it as part of CMSIS.

Reply:
And where do you want to have it ? People are getting CMSIS-DSP as part of CMSIS. They want to be able to compile it for host machines. So, the code has to support it.

Reply:
From my point of view, it would be helpful if Core had C fallbacks for its intrinsics.

Mbed OS uses Core + RTOS without DSP, and we have to worry about unit tests fallbacks for things like `__RBIT` ourselves.

Reply:
If we can have a cleaner solution answering the needs of everybody then I am all for it.

As usual, it is a matter of trade-off. We did the minimal fix in CMSIS-DSP to enable this feature (host compilation). It has been very well received by users of the library. 

Doing it cleanly is more work and not just impacting CMSIS-DSP.

Reply:
This is going off on a slight tangent, but as of GCC 9, it's possible to have a "best-of-both-worlds" for the cases where assembler is required for run-time speed.

You can use a C version for constant evaluation, and the assembler only for runtime, meaning the compiler can constant-evaluate things like `__RBIT(0)`. This previously was only possible when __RBIT was built-in.

See https://github.com/ARMmbed/mbed-os/pull/11897